### PR TITLE
Attendance Challenge and hackerrank exercises of day 3 

### PR DESCRIPTION
**Pull request by davidgzu**
[add] 
excercises of hacker Rank and challenge about heritance. polymorphism and  encapsulation

### CODE REVIEW

File: DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+public class PruebaFigura {
+    public static void main(String[] args) {
+
+        Figura[] figuras = new Figura[3]; // arreglo de referencias a figuras
+
+        figuras[0] = new Triangulo("morado", 4.3, 8.6);
+        figuras[1] = new Circulo("turquesa", 2.5);
+        figuras[2] = new Rectangulo("rosa", 15.2, 7.5);
+
+        for (int i = 0; i < 3; i++) {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java:10]: This is redundant. If you are defiying the array size as 3. just use figuras.length
Use meaningful var names **"i"**

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+public abstract class Figura {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:1]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    public double area;
+
+    public Figura() {
+    }
+
+    public Figura(String color) {
+        this.color = color;
+    }
+
+    public abstract void calcularArea();
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:12]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        this.radio = radio;
+    }
+
+    public void calcularArea() {
+        double area = PI * Math.pow(radio, 2);
+        this.area = area;
+        System.out.println("calculando area del circulo");
+    }
+
+    public void calcularDiametro() {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:21]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    public Circulo() {
+    }
+
+    public Circulo(String color, double radio) {
+        super(color);
+        this.radio = radio;
+    }
+
+    public void calcularArea() {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:15]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+public class Circulo extends Figura {
+
+    public static final double PI = 3.1416;
+
+    public double radio;
+
+    public Circulo() {
+    }
+
+    public Circulo(String color, double radio) {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:10]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+public class Circulo extends Figura {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:1]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+    public static final double PI = 3.1416;
+
+    public double radio;
+
+    public Circulo() {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:7]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public Circulo(String color, double radio) {
+        super(color);
+        this.radio = radio;
+    }
+
+    public void calcularArea() {
+        double area = PI * Math.pow(radio, 2);
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:16]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        this.area = area;
+        System.out.println("calculando area del circulo");
+    }
+
+    public void calcularDiametro() {
+        double diametro = radio * 2;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:22]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+        System.out.println("calculando diametro del circulo");
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:23]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        System.out.println(diametro);
+    }
+
+    public String toString() {
+        String estado = "Circulo - color: " + color + ", " + "radio: " + radio + ", " + "área: " + area;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circulo.java:28]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 13]
```diff
@@ -0,0 +1,14 @@
+public abstract class Figura {
+    public String color;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:2]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+    public double area;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:3]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+
+    public Figura() {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:5]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+    }
+
+    public Figura(String color) {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:8]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+        this.color = color;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Figura.java:9]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java [hunk 18]
```diff
@@ -0,0 +1,22 @@
+public class PruebaFigura {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java:1]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+    public static void main(String[] args) {
+
+        Figura[] figuras = new Figura[3]; // arreglo de referencias a figuras
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java:4]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+
+        figuras[0] = new Triangulo("morado", 4.3, 8.6);
+        figuras[1] = new Circulo("turquesa", 2.5);
+        figuras[2] = new Rectangulo("rosa", 15.2, 7.5);
+
+        for (int i = 0; i < 3; i++) {
+            if (figuras[i] instanceof Circulo) {
+                System.out.println("El objeto del indice " + i + " es un circulo.");
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java:12]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java [hunk 21]
```diff
@@ -0,0 +1,27 @@
+public class Rectangulo extends Figura {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java:1]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+
+    public double base;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java:3]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+    public double altura;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java:4]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java [hunk 24]
```diff
@@ -0,0 +1,27 @@
...
+    public Rectangulo(String color, double base, double altura) {
+        super(color);
+        this.base = base;
+        this.altura = altura;
+    }
+
+    public void calcularArea() {
+        double area = base * altura;
+        this.area = area;
+        System.out.println("Calculando el area del Rectangulo");
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java:18]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java [hunk 25]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public String toString() {
+        String estado = "Rectangulo - color: " + color + ", " + "base: " + base + ", " + "altura: " + altura + ", "
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Rectangulo.java:22]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java [hunk 26]
```diff
@@ -0,0 +1,28 @@
+public class Triangulo extends Figura {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java:1]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+
+    public double base;
+    public double altura;
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java:4]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+    public Triangulo(String color, double base, double altura) {
+        super(color);
+        this.base = base;
+        this.altura = altura;
+    }
+
+    public void calcularArea() {
+        double area = (base * altura) / 2;
+        this.area = area;
+        System.out.println("Calculando el area del triangulo");
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java:18]: Use English!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java [hunk 29]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public String toString() {
+        String estado = "Triangulo - color: " + color + ", " + "base: " + base + ", " + "altura: " + altura + ", "
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Triangulo.java:22]: Use English!

File: DavidGonzalezZumaya/Day3/HackerRank/OutputFormatting/OutputFormating.java [hunk 30]
```diff
@@ -0,0 +1,26 @@
...
+    public static void main(String[] args) {
+
+        Scanner scanner = new Scanner(System.in);
+
+        final String OUTPUT_FORMATTING = "%-15s%03d%n";
+        final String DECORATOR_FORMATTING = "================================";
+
+        System.out.println(DECORATOR_FORMATTING);
+
+        for (int i = 0; i < 3; i++) {
```
Comment (edbermeo) [inline DavidGonzalezZumaya/Day3/HackerRank/OutputFormatting/OutputFormating.java:14]: Declare 3 as a constant and use meaningful var names **"i"**

**Review by edbermeo — CHANGES_REQUESTED** [review]
Please use english everywhere! and check code formatting

File: DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java [hunk 31]
```diff
@@ -0,0 +1,22 @@
+public class PruebaFigura {
+    public static void main(String[] args) {
+
+        Figura[] figuras = new Figura[3]; // arreglo de referencias a figuras
+
+        figuras[0] = new Triangulo("morado", 4.3, 8.6);
```
Comment (luisalanmz) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/PruebaFigura.java:6]: Put color name in English and in a Constant

**Review by luisalanmz — CHANGES_REQUESTED** [review]

**Review by edbermeo — APPROVED** [review]
LGTM!

File: DavidGonzalezZumaya/Day3/ChallengeDay3/Circle.java [hunk 32]
```diff
@@ -0,0 +1,32 @@
+public class Circle extends Figure {
```
Comment (luisod78) [inline DavidGonzalezZumaya/Day3/ChallengeDay3/Circle.java:1]: The default unnamed package should not be used.
Try to arrange your classes into packages like:
com.bootcamp.dayone.excerciseone.....
or
com.inheritance.one.... two... etc

**Review by luisod78 — CHANGES_REQUESTED** [review]

**Review by luisod78 — APPROVED** [review]

**Pull request merged**